File: python/messages/point_message.py
import json
import numpy as np

class PointMessage:
    def __init__(self, position, normal):
        self.position = position
        self.normal = normal

    @staticmethod
    def from_json(json_string):
        point_message_dict = json.loads(json_string)
        return PointMessage(point_message_dict["position"], point_message_dict["normal"])

    def to_json(self):
        return json.dumps(self.__dict__)

File: Assets/Scripts/messages/PointMessage.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class PointMessage
{
    public Vector3 position;
    public Vector3 normal;

    public PointMessage(Vector3 position, Vector3 normal)
    {
        this.position = position;
        this.normal = normal;
    }

    public static PointMessage FromJson(string json)
    {
        return JsonUtility.FromJson<PointMessage>(json);
    }

    public string ToJson()
    {
        return JsonUtility.ToJson(this);
    }
}
